pkg/controller/atlasconnection/atlasconnection_controller.go

```diff
@@ -149,39 +149,43 @@ func (r *MongoDBAtlasConnectionReconciler) Reconcile(cx context.Context, req ctr
 
 	projectID := instance.InstanceInfo[dbaas.ProjectIDKey]
 
-	// Now create a configmap for non-sensitive information needed for connecting to the DB instance
-	cm := getOwnedConfigMap(conn, instance.InstanceInfo[dbaas.ConnectionStringsStandardSrvKey])
-	cmCreated, err := r.Clientset.CoreV1().ConfigMaps(req.Namespace).Create(context.Background(), cm, metav1.CreateOptions{})
-	if err != nil {
-		result := workflow.Terminate(workflow.MongoDBAtlasConnectionBackendError, err.Error())
-		dbaas.SetConnectionCondition(conn, dbaasv1alpha1.DBaaSConnectionProviderSyncType, metav1.ConditionFalse, string(result.Reason()), result.Message())
-		return ctrl.Result{}, fmt.Errorf("failed to create configmap:%w", err)
+	if conn.Status.ConnectionInfoRef == nil {
```

Comment:
If the secret and configmap references are already set, I think the code should try to verify or update the secret and configmap to make sure the secret and configmap are not deleted and the data are correct and unchanged. Another option is to use function `controllerutil.CreateOrUpdate` to create or update the secret and configmap.
But I don't think we have to do it in this PR.

Reply:
OK. Lets handle this kind of scenario in the future, given that  MongoDB does not have API to retrieve the password of a db user so there is no way to verify whether the data has changed or not.